google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -797,6 +819,24 @@ public Builder setDaysSinceCustomTime(Integer daysSinceCustomTime) {
           return this;
         }
 
+        /**
+         * Sets the list of prefixes. If any prefix matches the beginning of the object’s name, this
+         * portion of the condition is satisfied for that object.
+         */
+        public Builder setMatchesPrefix(List<String> matchesPrefix) {
+          this.matchesPrefix = matchesPrefix;
```

Comment:
For these setter methods, we'll want to guard the values passed in to be immutable values.
```suggestion
          this.matchesPrefix = matchesPrefix != null ? ImmutableList.copyOf(matchesPrefix) : null;
```